### DISCUSSION TITLE
disable "leave page" pop-up in pavlovia

### DISCUSSION THREAD

**Discussion by isaosenberger** [jspsych/jsPsych#2942] [Q&A] [answered]
Hi all,

I am using Pavlovia to run my JsPsych experiment. Everything is working, except for a pop-up window that I get whenever I leave my experiment page. I'd would be great if someone can help me disable that pop-up :)

For some context: in my experiment the participant gets the option to redirect to a different website with a button click. Whenever I (or the participants) press that button, I get this pop-up window asking me whether I really want to leave the experiment page:      
<img width="224" alt="leave page pop-up" src="https://user-images.githubusercontent.com/122807628/212713529-ef7e504a-ddca-43d4-8d72-58b696435979.png">

I'd like to disable that pop-up window, but I am unsure how to do that. I didn't find a way to disable the pop-up on Pavlovia, therefore I assume that I have to do that in my code. 

I did some searching and think that I found the source of my "problem" in the pavlovia plug-in, here:
```
// warn the user when they attempt to close the tab or browser:
const _beforeunloadCallback = (event) =>
{
// preventDefault should ensure that the user gets prompted:
event.preventDefault();
// Chrome requires returnValue to be set:
event.returnValue = '';
};
window.addEventListener('beforeunload', _beforeunloadCallback);
```

and here:
```
/**
* The callback for the beforeunload event, which is triggered when the participant
* tries to leave the experiment by closing the tab or browser.
*
* @type {null}
* @private
*/
PavloviaPlugin._beforeunloadCallback = null;
```

but I am unsure how to proceed from here. Any help is much appreciated!
I am using jspsych 7.2.3, and the pavlovia plug in is 2022.1.1

Thanks a lot!

Best,
Lisa

**Comment by jodeleeuw** [comment, marked as answer]
Hi @isaosenberger,

Can you modify the code to the pavlovia plugin? If so, removing the `window.addEventListener(...)` might do the trick.

If you can't easily modify the plugin code, then adding this statement might do the trick:

```js
window.addEventListener('beforeunload', function(event) {
    event.stopImmediatePropagation();
}, true);
```

This will add a new event listener to the `beforeunload` event, and it will stop the event from propogating so the other event listener for pavlovia will not be called. The downside is this will stop all functions tied to `beforeunload`, so you have to ensure that there are no other functions that are triggered by that event.

**Reply by isaosenberger** [reply]
Hi @jodeleeuw,

thanks for your help!

I commented out everything that has to do with the beforeunload event listener in the pavlovia plugin, and it works like a charm :)

Best,
Lisa
